monkestation/code/modules/antagonists/vampire/actions/innate/recuperation.dm

```diff
@@ -0,0 +1,282 @@
+	if(regrow_accumulation < 1)
+		return
+	regrow_accumulation--
+
+	if(regrow_limb_zone)
+		regrow_limb(regrow_limb_zone)
+	else
+		regrow_organ(regrow_organ_type)
+
+/// Gets a suitable limb zone for regrowth, if any.
+/datum/action/cooldown/vampire/recuperation/proc/get_regrow_limb_zone()
+	if(user.health < user.crit_threshold) // saves on having to check missing limbs
+		return
+
+	var/missing_zones = user.get_missing_limbs()
+	for(var/zone as anything in missing_zones)
+		var/obj/item/bodypart/limb = user.dna.species.bodypart_overrides[zone]
+		if(user.health - initial(limb.max_damage) < user.crit_threshold) // don't crit/kill the vampire if they regrow a limb
+			missing_zones -= zone
+
+	if(!length(missing_zones))
+		return
+
+	return pick(missing_zones)
+
+/// Gets a suitable organ type for regrowth, if any.
+/datum/action/cooldown/vampire/recuperation/proc/get_regrow_organ_type(regen_level)
+	if(regen_level < VAMPIRE_REGEN_THRESHOLD_REGROW_ORGANS)
+		return
+
+	// Unlike limb regrowth, this is in order of priority.
+	var/valid_slots = list(
+		ORGAN_SLOT_EYES, // seeing is the most important thing
+		ORGAN_SLOT_LUNGS, // right behind breathing
+		ORGAN_SLOT_EARS, // hearing you can go without for a bit
+		ORGAN_SLOT_LIVER, // a liver? definitely can go without
+		ORGAN_SLOT_HEART, // you can go without it entirely but fixing your tongue and ass before your heart would be fucking weird
+		ORGAN_SLOT_STOMACH, // your hunger is based on your lifeforce so this is useless too
+		ORGAN_SLOT_TONGUE, // next up, speaking properly
+		ORGAN_SLOT_BUTT, // clearly, farting is more important than pissing
+		ORGAN_SLOT_BLADDER
+	)
+
+	for(var/slot as anything in valid_slots)
+		if(user.get_organ_slot(slot))
+			continue
+
+		var/organ_type = user.dna.species.get_mutant_organ_type_for_slot(slot)
+		if(!organ_type)
+			continue
+
+		return organ_type
+
+/// Regrows a limb on the given zone.
+/datum/action/cooldown/vampire/recuperation/proc/regrow_limb(zone)
+	user.regenerate_limb(zone)
+
+	var/obj/item/bodypart/new_limb = user.get_bodypart(zone)
+	if(!new_limb) // i hope for the sake of this codebase that this hopefully pointless check never passes
+		CRASH("Vampire Recuperate attempted regrowing a limb using regenerate_limb(zone), yet get_bodypart(zone) returned null. Something is fucked up.")
+
+	new_limb.receive_damage(brute = new_limb.max_damage, forced = TRUE, wound_bonus = CANT_WOUND) // regrown limbs start at 0 health
+
+	playsound(user, 'sound/magic/demon_consume.ogg', vol = 50, vary = TRUE)
+	user.visible_message(
+		message = span_danger("[user]'s flesh shifts nauseatingly as [user.p_their()] [new_limb.plaintext_zone] regrows!"),
+		self_message = span_green("Your flesh shifts as your [new_limb.plaintext_zone] regrows!"),
+		blind_message = span_hear("You hear a wet crunch!")
+	)
+
+/// Regrows an organ of the given type.
+/datum/action/cooldown/vampire/recuperation/proc/regrow_organ(type)
+	var/obj/item/organ/organ = new type
```

Comment:
You might want to make sure 'type' is an organ

Reply:
The source of "type" is get_mutant_organ_type_for_slot, so I know for a fact that it's an organ type.
It's only a proc for code clarity purposes and is called from one place.